versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/IndexesLogicImpl.java

```diff
@@ -537,26 +540,37 @@ void completeIndexesInCommitChain(
 
       StoreIndex<CommitOp> newIndex;
       ObjId referenceIndex;
+      List<IndexStripe> indexStripes;
       if (parent != null) {
         newIndex = incrementalIndexForUpdate(parent, Optional.empty());
         referenceIndex = parent.referenceIndex();
+        indexStripes = parent.referenceIndexStripes();
       } else {
         newIndex = newStoreIndex(COMMIT_OP_SERIALIZER);
         referenceIndex = null;
+        indexStripes = Collections.emptyList();
       }
 
       commitOperations(current).forEach(newIndex::add);
 
-      CommitObj newCommit =
+      Builder c =
           commitBuilder()
               .from(current)
               .incompleteIndex(false)
               .referenceIndex(referenceIndex)
-              .incrementalIndex(newIndex.serialize())
-              .build();
-      newCommit = commitLogic.updateCommit(newCommit);
+              .referenceIndexStripes(indexStripes)
+              .incrementalIndex(newIndex.serialize());
+
+      if (parent != null) {
+        c.tail(Collections.singleton(parent.id()));
```

Comment:
Yes, that's logically correct.

I'd just propose to call `c.tail()` only once with a pre-populated list (otherwise immutables is a bit inefficient - think: list resizing).